package/MDAnalysis/auxiliary/base.py

```diff
@@ -0,0 +1,447 @@
+        represenatative.
+
+        Returns
+        -------
+        List (of length n_cols) of auxiliary value(s) 'representing' the 
+        timestep.
+        """
+
+        if self.cutoff != -1:
+            cutoff_data = np.array([self.ts_data[i] 
+                                    for i,d in enumerate(self.ts_diffs)
+                                    if d < self.cutoff])
+            cutoff_diffs = [d for d in self.ts_diffs if d < self.cutoff]
+        else:
+            cutoff_data = self.ts_data
+            cutoff_diffs = self.ts_diffs
+        if len(cutoff_data) == 0:
+            value = np.array([np.nan]*len(self.data_cols))
+        elif self.represent_ts_as == 'closest':
+            value = cutoff_data[np.argmin(cutoff_diffs),:]
+        elif self.represent_ts_as == 'average':
+            value = np.mean(cutoff_data, axis=0)
+        return value
+    
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.close()
+        return False
+
+    def close(self):
+        # Overwrite as appropriate
+        pass
+
+    @property
+    def time(self):
+        """ Time in ps of current auxiliary step.
+
+        As read from the appropriate column of the auxiliary data, if present; 
+        otherwise calcuated as ``step * dt + initial_time``
+        """
+        if self.time_col is not None:
+            return self._data[self.time_col]
+        else:
+            return self.step * self.dt + self.initial_time
+
+    @property
+    def step_data(self):
+        """ Auxiliary values of interest for the current step.
+
+        As taken from the appropariate columns (identified in `data_cols`) of 
+        the full auxiliary data read in for the current step.
+        """
+        return [self._data[i] for i in self.data_cols]
+
+    @property
+    def n_steps(self):
+        """ Total number of steps in the auxiliary data. """
+        try:
+            return self._n_steps
+        except AttributeError:
+            self._n_steps = self.count_n_steps()
+            return self._n_steps
+
+    @property
+    def times(self):
+        """ List of times of each step in the auxiliary data. 
+
+        Calculated using `dt` and `initial_time` if `constant_dt` is True; 
+        otherwise as read from each auxiliary step in turn. 
+        """
+          
+        try:
+            return self._times
+        except AttributeError:
+            if self.constant_dt:
+                self._times = [i*self.dt+self.initial_time 
+                               for i in range(self.n_steps)]
+            else:
+                self._times = self.read_all_times()
+            return self._times
+
+    @property
+    def dt(self):
+        """ Change in time between steps. 
+
+        Defaults to 1ps if not provided or read from auxiliary data. """
+        if self._dt:
+            return self._dt
+        else:
+            return 1  ## default to 1ps; WARN?
```

Comment:
In what case could that happen? Warning would be nice indeed, but could the issue get caught earlier?